src/__tests__/OPLSDA.test.js

```diff
@@ -19,34 +28,237 @@ labels=cbind(as.character(iris[,5]))
 model=opls(X, labels)
 */
 
-/*
-model@summary
-       R2X  R2Y   Q2
-PC_o 1 0.7 0.93 0.93
-*/
+const x = new Matrix(iris);
+const opls = new OPLS(x, metadata, { cvFolds: cvSets });
+describe('Statistic values with OPLS-DA working on iris', () => {
+  // > model@summary
+  //   R2X  R2Y   Q2 AUROC
+  // PC_o 1 0.72 0.67 0.67  0.67
+  it('Test statistic values', () => {
+    expect(opls.output.Q2y[0]).toBeCloseTo(0.6665861, 7);
+    expect(opls.output.Q2y[1]).toBeCloseTo(0.6665869, 7);
+    // Multi-class area under the curve: 0.6704
+    expect(opls.output.auc[0]).toBeCloseTo(0.6704, 4);
+    // Multi-class area under the curve: 0.6732
+    expect(opls.output.auc[1]).toBeCloseTo(0.6732, 4);
+    expect(opls.output.R2y[0]).toBeCloseTo(0.67, 2);
+    expect(opls.output.R2y[1]).toBeCloseTo(0.67, 2);
+  });
+
+  it('Test E matrix', () => {
+    //   > model@E
+    //       Sepal.Length Sepal.Width  Petal.Length  Petal.Width
+    //  [1,]  0.056367716  0.19711813 -2.417647e-05  0.061137059
+    //  [2,] -0.176379276 -0.77771969 -1.487345e-01 -0.128314741
+    //  [3,] -0.055799091 -0.24308369 -4.437359e-02 -0.041734675
+    //  [4,] -0.110771432 -0.36843166  1.732106e-02 -0.127486751
+    //  [5,]  0.113426110  0.47315503  7.063183e-02  0.093097059
+    const E = opls.output.E.to2DArray();
+    expect(E[0]).toBeDeepCloseTo(
+      [0.056367716, 0.19711813, -2.417647e-5, 0.061137059],
+      8,
+    );
+    expect(E[1]).toBeDeepCloseTo(
+      [-0.176379276, -0.77771969, -1.487345e-1, -0.128314741],
+      7,
+    );
+    expect(E[2]).toBeDeepCloseTo(
+      [-0.055799091, -0.24308369, -4.437359e-2, -0.041734675],
+      8,
+    );
+    expect(E[3]).toBeDeepCloseTo(
+      [-0.110771432, -0.36843166, 1.732106e-2, -0.127486751],
+      8,
+    );
+    expect(E[4]).toBeDeepCloseTo(
+      [0.11342611, 0.47315503, 7.063183e-2, 0.093097059],
+      8,
+    );
 
-test('Statistic values with OPLS-R working on iris dataset', () => {
-  const x = new Matrix(iris);
-  const y = newM.get('iris', { format: 'factor' }).values;
-  const opls = new OPLS(x, y, { cvFolds: kFoldStratified });
-  expect(opls.output.Q2y[0]).toBeCloseTo(0.93, 1);
-  expect(opls.output.Q2y[1]).toBeCloseTo(0.93, 1);
-  expect(opls.output.R2y[0]).toBeCloseTo(0.93, 1);
-  expect(opls.output.R2y[1]).toBeCloseTo(0.93, 1);
```

Comment:
I suggest you split tests for OPLS-DA and OPLS-regression in two different files, there is no reason to call it OPLSDA and then test for regression in it.

Reply:
Ok I agree this should be removed

Reply:
Ok